k6/docker-entrypoint.sh

```diff
@@ -0,0 +1,37 @@
+#!/bin/sh
+set -e
+
+# Check for the test case file via either a passed-in env path or the default
+# In either case, the file should be mounted as a volume (from configmap, etc)
+TEST_CASE_FILE=${TEST_CASE_FILE:-"/scripts/load_script.js"}
+TEST_CASE_OUTPUT=${TEST_CASE_OUTPUT:-"prometheus.txt"}
+
+if ! [ -f "${TEST_CASE_FILE}" ]; then
+        echo "A test case JS file must be provided." 1>&2
+        exit 1
+fi
+
+# Check that we have a Prometheus Pushgateway defined
+if [ -z "${PUSHGATEWAY_URL}" ]; then
+        echo "No Pushgateway URL" 1>&2
+				exit 1
+fi
+
+COMBINED_TEST_CASE_FILE="/home/k6/combined_load_script.js"
+
+# Check for a custom handleSummary() in the provided load_script, add it otherwise
+grep -q "export \+function \+handleSummary" ${TEST_CASE_FILE} && cp ${TEST_CASE_FILE} ${COMBINED_TEST_CASE_FILE} || cat ${TEST_CASE_FILE} handleSummary.js > ${COMBINED_TEST_CASE_FILE}
+
+# Run the test case. If we used the default handleSummary.js snippet, the output
+# file will be called prometheus.txt. If handleSummary was provided as part of
+# the test case js file, TEST_CASE_OUTPUT will need to be set to the corresponding output file namespace
+k6 run "${COMBINED_TEST_CASE_FILE}"
+
+# If
+if ! [ -f "${TEST_CASE_OUTPUT}" ]; then
+        echo "Test case output was not found. Unable to push to Prometheus." 1>&2
+        exit 1
+fi
+
+# Push to the Prometheus PushGateway
+cat ${TEST_CASE_OUTPUT} | curl --data-binary @- "${PUSHGATEWAY_URL}"
```

Comment:
This should be combined with the `-n "${PUSHGATEWAY_URL}"` to make sure there is a URL to push to.

Reply:
I tested for PUSHGATEWAY_URL lines 15 - 18 to exit right away on failure (and skip all further operations). Would you still prefer to wrap the PUSHGATEWAY_URL use attempt with a check?

Reply:
You should eliminate the check on 15-18 in favor of only sending the output if the push gateway URL is defined (i.e. the test didn't fail if couldn't report results; the _trial_ will still fail because it won't be able to collect those results later).

Reply:
Given that a load test could take many minutes to run, wouldn't it make sense to fail on no PUSHGATEWAY_URL before the load test begins so that the trial can fail faster and thus be quicker to debug?

Reply:
@jgustie can we resolve this?

Reply:
The other trial jobs will run without `PUSHGATEWAY_URL` set, this should be no different.

Reply:
Reworked check for PUSHGATEWAY_URL to check for successful test metric push if and only if PUSHGATEWAY_URL is not empty.

Reply:
@jgustie are we good here?